src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -324,6 +328,15 @@ public T[] ToArray()
             return arr;
         }
 
+        private void Grow()
+        {
+            int newcapacity = (int)((long)_array.Length * (long)GrowFactor / 100);
+            if (newcapacity < _array.Length + MinimumGrow)
+            {
+                newcapacity = _array.Length + MinimumGrow;
+            }
+            SetCapacity(newcapacity);
+        }
```

Comment:
Factoring this out is unrelated cleanup or is there a diff I'm missing? Generally, it's better to do these separately. I don't mind having this in a helper, just having it in this same diff makes it harder to review.

In terms of helpers, maybe add one for the pattern you now have in 3 places:  `void Increment(ref int index)`?

Also, I'd add a comment about not wanting to do modulus for perf reasons.